src/components/autocomplete/js/autocompleteController.js

```diff
@@ -64,6 +66,18 @@ function MdAutocompleteCtrl ($scope, $element, $mdUtil, $mdConstant, $mdTheming,
     $mdUtil.initOptionalProperties($scope, $attrs, { searchText: null, selectedItem: null });
     $mdTheming($element);
     configureWatchers();
+
+    if ($scope.componentId) {
+      removeFromRegistry = $mdComponentRegistry.register({
+        // Restricted API interface for the component registry.
+        // We don't want to provide the whole controller as API.
+        clearCache: function() {
+          // Run the clearCache method in the controller context.
+          ctrl.clearCache();
```

Comment:
Our code shouldn't be written to make jasmine work, it supposed to be opposite. 
I think the component registry is important and i do think there're ways to bypass that